### PR TITLE
[api-draft#1990][Common] The DeployMode code cleanup

### LINKED ISSUE #1990: [Feature][Common] The DeployMode code cleanup
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

The DeployMode code cleanup,

The DeployMode property judgment is performed in the CommandBuilder class. I think this step is repetitive, because when the CommandArgs parameter is generated, there will be a converter to convert it into an enumeration value.

https://github.com/apache/incubator-seatunnel/blob/ede29370f7571cbf140533901213c0d679702bc3/seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/command/SparkCommandBuilder.java#L29-L32

https://github.com/apache/incubator-seatunnel/blob/ede29370f7571cbf140533901213c0d679702bc3/seatunnel-core/seatunnel-spark-starter/src/main/java/org/apache/seatunnel/core/starter/spark/args/SparkCommandArgs.java#L29-L33

https://github.com/apache/incubator-seatunnel/blob/ede29370f7571cbf140533901213c0d679702bc3/seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/DeployModeConverter.java#L30-L33

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhuangchong**
## Purpose of this pull request

this pr closes #1990 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-common/src/main/java/org/apache/seatunnel/common/config/Common.java [hunk 1]
```diff
@@ -70,19 +54,17 @@ public static Optional<String> getDeployMode() {
...
                 return Paths.get(path).getParent().getParent();
             } catch (URISyntaxException e) {
                 throw new RuntimeException(e);
             }
-        } else if (MODE.equals(Optional.of(DeployMode.CLUSTER.getName()))) {
-            return Paths.get("");
-        } else {
-            throw new IllegalStateException("MODE not support : " + MODE.orElse("null"));
         }
+        // deploy mode: cluster
```
Comment (Hisoka-X) [inline seatunnel-common/src/main/java/org/apache/seatunnel/common/config/Common.java]: Please cover this logic in if statement, if we have new another DeployMode then throw not support exception

Comment (zhuangchong) [reply seatunnel-common/src/main/java/org/apache/seatunnel/common/config/Common.java]: done.

**Review by Hisoka-X — APPROVED** [review]
LGTM

**Comment by zhuangchong** [comment]
> LGTM

@BenJFan thanks.

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**